src/ping.c

```diff
@@ -346,67 +386,82 @@ static void *ping_thread(void *arg) /* {{{ */
   return (void *)0;
 } /* }}} void *ping_thread */
 
-static int start_thread(void) /* {{{ */
+
+static int stop_thread(void) /* {{{ */
 {
-  int status;
+  int status = 0;
 
   pthread_mutex_lock(&ping_lock);
 
-  if (ping_thread_loop != 0) {
-    pthread_mutex_unlock(&ping_lock);
-    return 0;
-  }
-
-  ping_thread_loop = 1;
-  ping_thread_error = 0;
-  status = plugin_thread_create(&ping_thread_id, ping_thread,
-                                /* arg = */ (void *)0, "ping");
-  if (status != 0) {
-    ping_thread_loop = 0;
-    ERROR("ping plugin: Starting thread failed.");
+  if (ping_thread_loop == 0) {
     pthread_mutex_unlock(&ping_lock);
     return -1;
```

Comment:
Sorry for commenting on pre-existing code, but this really should return zero.

```suggestion
    return 0;
```